core/src/main/scala/kafka/log/OffsetMap.scala

```diff
@@ -172,6 +298,33 @@ class SkimpyOffsetMap(val memory: Int, val hashAlgorithm: String = "MD5") extend
     lastOffset = offset
   }
 
+  /**
+   * Search for the hash of the key by repeated probing until we find the hash we are looking for or we find an empty slot
+   * @param key The key
+   * @return true if key found otherwise false
+   */
+  private def search(key: ByteBuffer): Boolean = {
```

Comment:
Also add in java doc that when the search returns true, the position of the bytebuffer should be pointing to the starting position of the offset for that entry.